components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -1066,6 +1066,42 @@ public API getAPI(APIIdentifier identifier,APIIdentifier oldIdentifier, String o
         }
     }
 
+    @Override
+    public Set<Tier> getAllTiers() throws APIManagementException {
+        Set<Tier> tiers = new TreeSet<Tier>(new TierNameComparator());
+
+        Map<String, Tier> tierMap;
+        if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
+            tierMap = APIUtil.getAllTiers();
+        } else {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId, true);
+            tierMap = APIUtil.getAllTiers(tenantId);
+            PrivilegedCarbonContext.endTenantFlow();
```

Comment:
startTenantFlow need to be in try block and endTenantFlow should done in finally block